### PR TITLE
BLADERUNNER: added basic support for blade runner game

### PR DESCRIPTION
**Pull request by peterkohaut**
current status:
game is completely unplayable, but engine can load and render game scenes and actors (you can walk within and between first two scenes of game).

what is implemented:
- basic structures from game engine are implemented
- scenes can be loaded and theirs scripts are fully decompiled
- decoding and rendering of vqa files (both video and audio)
- rendering of actors and scenes (it is not using polygons nor voxels for rendering items and actors)
- basic walking of main actor is working
- about 50% of api for game scripts is implemented 

what is missing:
- ai script (for every actor there is different script in total of 74) - these are needed because these contains logic for walking and animations of characters
- ui for esper, kia, vk, spinners, elevator, clues databases 
- walking with pathfinding
- dialogs
- combat
- rendering of items and shadows
- saving and loading games

### CODE REVIEW

**Comment by wjp** [comment]
Great news, and this looks like an impressive amount of work went into it, but could you please provide a description for this pull request, containing at least what the status of this is?

**Comment by peterkohaut** [comment]
current status:
game is completely unplayable, but engine can load and render game scenes and actors (you can walk within and between first two scenes of game).

what is implemented:
- basic structures from game engine are implemented
- scenes can be loaded and theirs scripts are fully decompiled
- decoding and rendering of vqa files (both video and audio)
- rendering of actors and scenes
- basic walking of main actor is working
- about 50% of api for game scripts is implemented 

what is missing:
- ai script (for every actor there is different script in total of 74) - these are needed because these contains logic for walking and animations of characters
- ui for esper, kia, vk, spinners, elevator, clues databases 
- walking with pathfinding
- dialogs
- combat
- rendering of items and shadows
- saving and loading games

**Comment by wjp** [comment]
Your commit messages tend to have very long lines. If you look at https://github.com/scummvm/scummvm/pull/836/commits , you'll notice they often don't fit. (Also see http://wiki.scummvm.org/index.php/Commit_Guidelines )

File: engines/bladerunner/item_pickup.cpp [hunk 1]
```diff
@@ -95,7 +95,7 @@ void ItemPickup::tick() {
 		_facing -= float(2.0f * M_PI);
 	}
 
-	_animationFrame = (_animationFrame + 1) % _vm->_sliceAnimations->getFrameCount(_animationId);
+	_animationFrame = ((int)(_animationFrame + 1)) % ((int)_vm->_sliceAnimations->getFrameCount(_animationId));
```
Comment (wjp) [inline engines/bladerunner/item_pickup.cpp]: Would it make sense to make `getFrameCount` return uint32 instead of float?

**Pull request merged**